### DISCUSSION TITLE
ValueError: Can't clean for JSON: [0.0, 0.0, 8.508049964904785]

### DISCUSSION THREAD

**Discussion by Sapta0802** [NVlabs/sionna#452] [General]
![Screenshot 2024-05-29 101429](https://github.com/NVlabs/sionna/assets/88283648/29de53b7-0e1b-4f47-8d43-e080fc552262)
I'm experiencing an issue when trying to preview an XML file from the Blender Mitsuba add-on in Google Colab using the Sionna library. The error message I receive is: ValueError: Can't clean for JSON: [0.0, 0.0, 8.508049964904785],  Any guidance on how to resolve this.

**Comment by merlinND** [comment]
Hello @Sapta0802,

Does the scene render correctly with Mitsuba, not using Sionna? This will help check whether the problem comes from the scene or Sionna.
Please also share the scene file (including all meshes and textures) so that we can try reproducing locally.

**Reply by Sapta0802** [reply]
Thanks for your reply.

Mitsuba Rendering: I checked the scene by rendering it directly with Mitsuba, and it renders correctly. 

Scene Files: I have uploaded the scene file, including all meshes and textures.
[untitled_2.zip](https://github.com/NVlabs/sionna/files/15481523/untitled_2.zip)
[meshes.zip](https://github.com/NVlabs/sionna/files/15481528/meshes.zip)

mainly this problem happens when I am using google colab.

**Reply by merlinND** [reply]
This is an issue from `ipywidgets` / `traitlets` which has already been fixed upstream (see e.g. https://github.com/jupyter-widgets/pythreejs/issues/92).
The issue is that Google Colab seems to require a specific `ipython` and widgets version that does not include the fix. I tried selectively updating some packages, but couldn't find a combination that works.
Locally, everything works fine for me when using the latest `ipython`, `jupyterlab` and `pythreejs` versions.

I don't think there's anything we can do until Colab updates its `ipython` dependency.

**Reply by merlinND** [reply]
Edit: after searching on StackOverflow, I found a workaround for the IPython incompatibility, which allows updating IPython on your own in Colab.

Make the first cell of you Colab notebook the following:
```py
!pip install ipython tornado ipywidgets traitlets --upgrade

# Source: https://stackoverflow.com/q/76837647
# There has got to be a better way to do this...but there's a bad import in some of the colab files
# IPython no longer exposes traitlets like this, it's a separate package now
!sed -i "s/from IPython.utils import traitlets as _traitlets/import traitlets as _traitlets/" /usr/local/lib/python3.10/dist-packages/google/colab/*.py
!sed -i "s/from IPython.utils import traitlets/import traitlets/" /usr/local/lib/python3.10/dist-packages/google/colab/*.py

!pip install sionna
```

Once it is done executing, restart the runtime and the scene preview should now work:
```py
import sionna
scene = sionna.rt.load_scene('/content/scene/untitled_2.xml')
scene.preview()
```

![Screenshot from 2024-05-31 12-41-28](https://github.com/NVlabs/sionna/assets/5563370/71799f88-809e-418e-9251-5a1879d063e2)

**Reply by JNUshuaibi** [reply]
> 您好 ，
> 
> 场景是否使用 Mitsuba 正确渲染，而不是使用 Sionna？这将有助于检查问题是来自 scene 还是 Sionna。请同时共享场景文件（包括所有网格和纹理），以便我们可以尝试在本地重现。

When I use Jupyter Notebook, I encountered this error. How should I resolve it?

**Reply by merlinND** [reply]
Hello @JNUshuaibi,

Did you try the solution described above in this thread?

If you're using a local Jupyter Notebook (not Colab), then the first thing to try would be to upgrade the relevant packages:

```bash
python3 -m pip install --upgrade jupyter jupyterlab ipython pythreejs ipywidgets traitlets
```

**Reply by JNUshuaibi** [reply]
> 您好 ，
> 
> 您是否尝试过此线程中上述解决方案？
> 
> 如果您使用的是本地 Jupyter Notebook（而不是 Colab），那么首先要尝试的是升级相关软件包：
> 
> ```shell
> python3 -m pip install --upgrade jupyter jupyterlab ipython pythreejs ipywidgets traitlets
> ```
Hello!
After setting up a transmitter, check cover_map and find that the transmitter is aligned in a certain direction, but what I want is how to adjust the omnidirectional transmitter?

![118](https://github.com/user-attachments/assets/32f642a8-5fdd-442f-a43d-9509da31fe53)
here's my code:

```py
scene.tx_array = PlanarArray(num_rows=1,
                num_cols=1,
                vertical_spacing=0.5,
                horizontal_spacing=0.5,
                pattern="tr38901",
                polarization="V")
scene.rx_array = PlanarArray(num_rows=1,
                num_cols=1,
                vertical_spacing=0.5,
                horizontal_spacing=0.5,
                pattern="dipole",
                polarization="cross")
tx = Transmitter("tx", [4, 2, 20], [0, 0, 0])
scene.add(tx)

cm = scene.coverage_map(max_depth=5,
                        diffraction=True, # Disable to see the effects of diffraction
                        cm_cell_size=(5., 5.), # Grid size of coverage map cells in m
                        combining_vec=None,
                        precoding_vec=None,
                        num_samples=int(20e6)) # Reduce if your hardware does not have enough memory


# Create new camera
tx_pos = scene.transmitters["tx"].position.numpy()
bird_pos = tx_pos.copy()
bird_pos[-1] = 1000 # Set height of coverage map to 1000m above tx
bird_pos[-2] -= 0.01 # Slightly move the camera for correct orientation

# Create new camera
bird_cam = Camera("birds_view", position=bird_pos, look_at=tx_pos)

scene.add(bird_cam)

if no_preview:
    scene.render(camera="birds_view", coverage_map=cm, num_samples=512, resolution=resolution);
cm.show()
```

**Reply by merlinND** [reply]
Hello @JNUshuaibi,

This is unrelated to the current discussion, please open a new discussion and ask there.
